integration/test/performancetest/transmitter.go

```diff
@@ -153,12 +197,194 @@ func (transmitter *TransmitterAPI) TableExist() (bool, error) {
+	return newAttributes, nil
+}
+/*
+UpdateItem()
+Desc:
+	This function updates the item in dynamo if the atomic condition is true else it will return ConditionalCheckFailedException 
+Params:
+	hash: this is the commitHash
+	targetAttributes: this is the targetAttribute to be added to the dynamo item
+	testHash: this is the hash of the last item, used like a version check
+*/
+func (transmitter * TransmitterAPI) UpdateItem(hash string,targetAttributes map[string]types.AttributeValue) error{
+	var err error
+	var ae *types.ConditionalCheckFailedException // this exception represent the the atomic check has failed
+	fmt.Println("Updating:",hash)
+	item,err := transmitter.Query(hash) // get most Up to date item from dynamo | O(1) bcs of global sec. idx.
+	if len(item) ==0{ // check if hash is in dynamo
+		return errors.New("ERROR: Hash is not found in dynamo")
+	}
+	commitDate := fmt.Sprintf("%d",int(item[0]["CommitDate"].(float64)))
+	year := fmt.Sprintf("%d",int(item[0]["Year"].(float64)))
+	testHash := item[0][TEST_HASH].(string)
+	//setup the update expression
+	expressionAttributeValues := make(map[string]types.AttributeValue)
+	expression := ""
+	n_expression := len(targetAttributes)
+	i :=0
+	for attribute, value := range targetAttributes{
+		expressionName := ":" +strings.ToLower(attribute)
+		expression = fmt.Sprintf("set %s = %s",attribute,expressionName)
+		expressionAttributeValues[expressionName] = value
+		if(n_expression -1 >i){
+			expression += "and"
+		}
+		i++
+	}
+	expressionAttributeValues[":testHash"] = &types.AttributeValueMemberS{Value: testHash}
+	//----
+	//call update
+	_, err = transmitter.dynamoDbClient.UpdateItem(context.TODO(), &dynamodb.UpdateItemInput{
+        TableName: aws.String(transmitter.DataBaseName),
+        Key: map[string]types.AttributeValue{
+            "Year": &types.AttributeValueMemberN{Value: year},
+			"CommitDate": &types.AttributeValueMemberN{Value: commitDate },
+        },
+        UpdateExpression: aws.String(expression),
+        ExpressionAttributeValues: expressionAttributeValues,
+		ConditionExpression: aws.String("#testHash = :testHash"),
+		ExpressionAttributeNames: map[string]string{
+			"#testHash": TEST_HASH,
+		},
+    })
 
+    if err != nil && !errors.As(err,&ae) {
+        panic(err)
+    }
+	return err
+}
+/*
+UpdateReleaseTag()
+Desc: This function takes in a commit hash and updates the release value to true 
+Param: commit hash in terms of string 
+*/
+func (transmitter * TransmitterAPI) UpdateReleaseTag(hash string) error{
+	var err error
+	var ae *types.ConditionalCheckFailedException
+	attributes := map[string]types.AttributeValue{
+		"isRelease":&types.AttributeValueMemberBOOL{Value: true},
+	}
+	for {//concurrency retry
+		/*solving parallel updates using an optimistic lock and retry,
+		this may result in a livelock;
+		however, random sleeps makes this possiblity nearly impossible*/
+		err = transmitter.UpdateItem(hash,attributes) //try to update the item
```

Comment:
For metrics, I understand there is a need for `optimistic lock`. However, would it be the same when updating the `isRelease` flag? Do I care the second attemp to update to `true` if the first attempt update success to `true` ? So seems like `pessimistic lock` is more useful for me in this case.

Reply:
So dynamo doesn't have pessimistic locks, and I have that optimistic lock there because if we update release the moment the test ends, the isRelease flag can get overwritten (I had this bug while running previous releases).

Reply:
I'm not sure the use case here. Why we need to update the release the moment the test ends though? There would be a span amount of time in merging PR and deploying new version so I'm still hard to understand why we need `optimistic lock`?

Reply:
It doesn't have to end at the same time. So when a release is published this function is called in order to mark the (already existing at that point) commit the release is connected to as isReleased. Since the commit is already existing I need to use the update function. The optimistic lock is there to prevent if a releases happened while performance test is still uploading. This is an edge case but an edge case I ran it to when I was running this for previous release, where I updated releases right after the performance test. In addition to overcoming this edge case, this optimistic lock also ensures that we will never overwrite an commit's data. You can imagine this as a safety feature and--from my understanding--an industry standard practice.